site/docs/serialization/binary_serialization.md

```diff
@@ -1,5 +1,63 @@
+
+=== "Plan Message"
+
+    ```proto
+%%% proto.message.Plan %%%
+    ```
+
+## Extensions
+Protobuf supports both [simple](/extensions/#simple-extensions) and [advanced](/extensions/#advanced-extensions) extensions. Simple extensions are declared at the plan level and advanced extensions are declared at multiple levels of messages within the plan.
+
+### Simple Extensions
+
+For simple extensions, a plan references the URIs associated with the simple extensions to provide additional plan capabilities. These URIs will list additional relevant information for the plan. 
+
+Since references to simple extensions are used extensively in most plans, the serialized representation is normalized within the plan message type minimize space and serialization costs. For each simple extension YAML referenced, the plan will defined an ExtensionURI object and in-plan surrogate key that can be used in each simple extension extension definition. 
+
+=== "Advanced Extension Protobuf"
+    ```proto
+%%% proto.message.SimpleExtensionURI %%%
+    ```
+
+Once the YAML file is defined with a surrogate key, the surrogate key will be referenced by zero or more `SimpleExtensionDefinition`s. For each simple extension definition, a surrogate key is defined for that specific extension entity. This surrogate key is then pointed to within lower-level primitives (functions, etc) to reference that specific extension. Message properties are named `*_surrogate_key` where the surrogate key is defined and `*_pointer` when referencing the surrogate key. For example `function_surrogate_key` and `function_pointer`.
+
+=== "Advanced Extension Protobuf"
+    ```proto
+%%% proto.message.SimpleExtensionDefinition %%%
+    ```
+
+!!! note
+  Surrogate keys only have meaning within a single plan for the purposes of plan size reduction. They are not some form of global identifier. Different plans may use different surrogate keys for the same specific functions, types, type variations, etc.
+
+!!! note
+  It is valid for a plan to include `SimpleExtensionURI`s and/or `SimpleExtensionDefinition`s that are not referenced directly. It is an optimization to keep this list to the minimum set required.
+
+
+
+### Advanced Extensions
+
+Substrait protobuf exposes a special object in multiple places in the representation to expose extension capabilities. Extensions are done via this object. Extensions are separated into main concepts: 
+
+| Advanced Extension Type | Description                                                  |
+| ----------------------- | ------------------------------------------------------------ |
+| Optimization            | A change to the plan that may help some consumers work more efficiently with the plan. These properties should be propagated through plan pipelines where possible but do not impact the meaning of the plan. A consumer can safely ignore some/all of these properties. |
```

Comment:
```suggestion
| Optimization            | A change to the plan that may help some consumers work more efficiently with the plan. These properties should be propagated through plan pipelines where possible but do not impact the meaning of the plan. A consumer can safely ignore these properties. |
```

I'm not sure why you said *some*/all.  Are there examples of optimization extensions that can't be ignored?